sale_commission_product_criteria_domain/models/partner.py

```diff
@@ -40,14 +40,19 @@ def _onchange_agent_ids(self):
         for rec in self:
             exiting_agents = rec.commission_item_agent_ids.mapped("agent_id")
             to_create = [
-                {"partner_id": rec._origin.id, "agent_id": x._origin.id}
+                {
+                    "partner_id": rec._origin.id,
+                    "agent_id": x._origin.id,
+                    "group_ids": [(6, 0, x.allowed_commission_group_ids.ids)],
+                }
                 for x in rec.agent_ids.filtered(
                     lambda x: x.commission_id.commission_type == "product_restricted"
                 )
-                if x not in exiting_agents.ids
+                if x._origin.id not in exiting_agents.ids
```

Comment:
```suggestion
                if x not in exiting_agents
```
I think this would be easier and more readable, unless it doesn't work for some reason?

Reply:
No it does not work, since x is a new instance without id.